server/plugin/plugin.go

```diff
@@ -173,6 +174,12 @@ func (p *Plugin) OnConfigurationChange() (err error) {
 		if err != nil {
 			e.Logger.Errorf("Error updating job manager with config. err=%v", err)
 		}
+
+		telemetryClient, err := telemetry.NewRudderClient()
+		if err != nil {
+			e.Logger.Errorf("Cannot create telemetry client. err=%v", err)
```

Comment:
Should the code `return` here to ensure `e.Tracker` doesn't get overwritten?

Reply:
This actually can be done completely different. I originally created the client here, since I thought I needed more variables to create the client (that may change on configuration change). Nevertheless, since the client is always the same, we could declare the client somewhere else, and just write the tracker.

Regarding having the tracker getting overwritten, my original idea is that if there is an error with the creation of the client, the telemetryClient variable would be nil. Creating a tracker with a nil client will just create a tracker that doesn't track, making a silent error that admins can deal with (or not) without interfering with the plugin behaviour. We also want to update the tracker either way, because we might have changed the "EnableDiagnostics" variable, and we want to stop either way the tracking. If we do not update the Tracker, we will be using the "old client" to track things even if the EnableDiagnostics have been changed to false.

Reply:
Got it. Thanks for the detailed explanation :+1: